### PR TITLE
GEN-2470 - refact(MixedBreedPicker): migrate to vanilla extract css

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

MixedBreedPicker: migrate styles to vanilla extract css

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4588?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#4588** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4588?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#4586** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4586?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @guilhermespopolin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by alebedev — APPROVED** [review]

File: apps/store/src/components/PriceCalculator/MixedBreedPicker/MixedBreedPicker.tsx [hunk 1]
```diff
@@ -45,106 +44,49 @@ export const MixedBreedPicker = ({
...
+        className={combobox}
+        items={availableBreeds}
+        displayValue={(breed) => breed.displayName}
+        selectedItem={null}
+        onSelectedItemChange={handleAdd}
+        placeholder={t('FIELD_MIXED_BREEDS_PLACEHOLDER')}
+        noMatchesMessage={t('FIELD_BREEDS_NO_OPTIONS')}
+        mutliSelect={true}
+        disabled={loading}
+        required={required && selectedBreeds.length === 0}
```
Comment (alebedev) [inline apps/store/src/components/PriceCalculator/MixedBreedPicker/MixedBreedPicker.tsx:87]: Do we have a reason to remove required when something is selected? I would hope it just works without us spood-feeding the knowledge that non-empty value means requried field is now valid

**Review by gustaveen — APPROVED** [review]
👌

File: apps/store/src/components/PriceCalculator/MixedBreedPicker/MixedBreedPicker.tsx (hunk 1 above)
Comment (guilhermespopolin) [reply apps/store/src/components/PriceCalculator/MixedBreedPicker/MixedBreedPicker.tsx:87]: Good point.

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Aug 14, 8:12 AM EDT**: @guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/4588).

File: apps/store/src/components/PriceCalculator/MixedBreedPicker/MixedBreedPicker.tsx (hunk 1 above)
Comment (alebedev) [reply apps/store/src/components/PriceCalculator/MixedBreedPicker/MixedBreedPicker.tsx]: Agree, I see no reason to control `required` from the outside

**Pull request merged**